dev-libs/libuv/patches/libuv-1.21.0.patchset

```diff
@@ -0,0 +1,496 @@
+ LT_INIT
+ # TODO(bnoordhuis) Check for -pthread vs. -pthreads
+ AC_CHECK_LIB([dl], [dlopen])
++AC_CHECK_LIB([root], [dlopen])
+ AC_CHECK_LIB([kstat], [kstat_lookup])
+ AC_CHECK_LIB([nsl], [gethostbyname])
++AC_CHECK_LIB([network], [gethostbyname])
+ AC_CHECK_LIB([perfstat], [perfstat_cpu])
+ AC_CHECK_LIB([pthread], [pthread_mutex_init])
+ AC_CHECK_LIB([rt], [clock_gettime])
++AC_CHECK_LIB([root], [clock_gettime])
+ AC_CHECK_LIB([sendfile], [sendfile])
+ AC_CHECK_LIB([socket], [socket])
++AC_CHECK_LIB([network], [socket])
+ AC_SYS_LARGEFILE
+ AM_CONDITIONAL([AIX],      [AS_CASE([$host_os],[aix*],          [true], [false])])
+ AM_CONDITIONAL([ANDROID],  [AS_CASE([$host_os],[linux-android*],[true], [false])])
+@@ -57,6 +62,7 @@ AM_CONDITIONAL([DARWIN],   [AS_CASE([$host_os],[darwin*],       [true], [false])
+ AM_CONDITIONAL([DRAGONFLY],[AS_CASE([$host_os],[dragonfly*],    [true], [false])])
+ AM_CONDITIONAL([FREEBSD],  [AS_CASE([$host_os],[*freebsd*],     [true], [false])])
+ AM_CONDITIONAL([LINUX],    [AS_CASE([$host_os],[linux*],        [true], [false])])
++AM_CONDITIONAL([HAIKU],    [AS_CASE([$host_os],[haiku*],        [true], [false])])
+ AM_CONDITIONAL([MSYS],     [AS_CASE([$host_os],[msys*],         [true], [false])])
+ AM_CONDITIONAL([NETBSD],   [AS_CASE([$host_os],[netbsd*],       [true], [false])])
+ AM_CONDITIONAL([OPENBSD],  [AS_CASE([$host_os],[openbsd*],      [true], [false])])
+@@ -66,6 +72,9 @@ AM_CONDITIONAL([WINNT],    [AS_CASE([$host_os],[mingw*],        [true], [false])
+ AS_CASE([$host_os],[mingw*], [
+     LIBS="$LIBS -lws2_32 -lpsapi -liphlpapi -lshell32 -luserenv -luser32"
+ ])
++AS_CASE([$host_os],[haiku*], [
++    LIBS="$LIBS -lroot -lbe -lnetwork -lshared"
++])
+ AS_CASE([$host_os], [netbsd*], [AC_CHECK_LIB([kvm], [kvm_open])])
+ AC_CHECK_HEADERS([sys/ahafs_evProds.h])
+ AC_CONFIG_FILES([Makefile libuv.pc])
+diff --git a/include/uv/errno.h b/include/uv/errno.h
+index 8eeb95d..00d6d5c 100644
+--- a/include/uv/errno.h
++++ b/include/uv/errno.h
+@@ -248,7 +248,7 @@
+ # define UV__ENOENT (-4058)
+ #endif
+ 
+-#if defined(ENOMEM) && !defined(_WIN32)
++#if defined(ENOMEM) && !defined(_WIN32) && !defined(__HAIKU__)
+ # define UV__ENOMEM UV__ERR(ENOMEM)
+ #else
+ # define UV__ENOMEM (-4057)
+diff --git a/include/uv/unix.h b/include/uv/unix.h
+index 74a0d64..3fc8229 100644
+--- a/include/uv/unix.h
++++ b/include/uv/unix.h
+@@ -62,7 +62,7 @@
+       defined(__OpenBSD__)         || \
+       defined(__NetBSD__)
+ # include "uv/bsd.h"
+-#elif defined(__CYGWIN__) || defined(__MSYS__)
++#elif defined(__CYGWIN__) || defined(__MSYS__) || defined(__HAIKU__)
+ # include "uv/posix.h"
+ #endif
+ 
+diff --git a/src/unix/bsd-ifaddrs.c b/src/unix/bsd-ifaddrs.c
+index 0d02154..d0b3ba1 100644
+--- a/src/unix/bsd-ifaddrs.c
++++ b/src/unix/bsd-ifaddrs.c
+@@ -25,12 +25,20 @@
+ #include <errno.h>
+ #include <stddef.h>
+ 
++#if !defined(__HAIKU__)
+ #include <ifaddrs.h>
++#else
++#include <bsd/ifaddrs.h>
```

Comment:
The `bsd` subfolder is a part of the search path, and it shouldn't be specified explicitly, or bad things may happens. See #1882